### PR TITLE
Make sheet loading sync.

### PR DESCRIPTION
**Pull request by frzyc**
Since asset was moved away from the sheets, there is no need to have sheets load async, since it makes a lot of the front-end logic complex.

- Make sync loading of sheets
- Artifacts validate set/rarity/level using sheets
- character validate weapon types ( no more dullblade as default weapon for every char)
- Weapon validate key/level
- Refactor/deprecate consts

### CODE REVIEW

**Comment by frzyc** [comment]
could probably afford to get rid of all the `useMemo` when accessing sheets, since its just a lookup now instead of a Promise

File: apps/frontend/src/app/Components/Character/LocationAutocomplete.tsx [hunk 1]
```diff
@@ -33,14 +32,14 @@ export function LocationAutocomplete({ location, setLocation, filter = () => tru
...
+  ...Array.from(new Set(database.chars.keys.filter(k => getCharSheet(k, gender) ? filter(getCharSheet(k, gender)) : true).map(k => charKeyToLocCharKey(k))))
     .map(v => ({ key: v, label: toText(v), favorite: isFavorite(v) }))
     .sort((a, b) => {
       if (a.favorite && !b.favorite) return -1
       if (!a.favorite && b.favorite) return 1
       return a.label.localeCompare(b.label)
     })
-  ], [t, toText, isFavorite, database, characterSheets, filter, gender])
+  ], [t, toText, isFavorite, database, filter, gender])
   return <Suspense fallback={<Skeleton variant="text" width={100} />}>
```
Comment (nguyentvan7) [inline apps/frontend/src/app/Components/Character/LocationAutocomplete.tsx:43]: Another possible suspense removal

File: apps/frontend/src/app/Components/Weapon/WeaponCardNano.tsx [hunk 2]
```diff
@@ -25,7 +25,7 @@ type Data = {
 export default function WeaponCardNano({ weaponId, showLocation = false, onClick, BGComponent = CardDark, }: Data) {
   const { database } = useContext(DatabaseContext)
   const weapon = useWeapon(weaponId)
-  const weaponSheet = usePromise(() => weapon?.key && WeaponSheet.get(weapon.key), [weapon?.key])
+  const weaponSheet =  weapon?.key && getWeaponSheet(weapon.key)
```
Comment (nguyentvan7) [inline apps/frontend/src/app/Components/Weapon/WeaponCardNano.tsx:28]: 2 spaces :)

File: apps/frontend/src/app/Components/Weapon/WeaponCardPico.tsx [hunk 3]
```diff
@@ -1,19 +1,19 @@
...
-import usePromise from '../../ReactHooks/usePromise';
 import useWeapon from '../../ReactHooks/useWeapon';
 import CardDark from '../Card/CardDark';
 import SqBadge from '../SqBadge';
 import WeaponNameTooltip from './WeaponNameTooltip';
 
 export default function WeaponCardPico({ weaponId }: { weaponId: string }) {
   const weapon = useWeapon(weaponId)
-  const weaponSheet = usePromise(() => weapon?.key && WeaponSheet.get(weapon.key), [weapon?.key])
+  const weaponSheet =  weapon?.key && getWeaponSheet(weapon.key)
```
Comment (nguyentvan7) [inline apps/frontend/src/app/Components/Weapon/WeaponCardPico.tsx:16]: 2 space

File: apps/frontend/src/app/Types/consts.ts [hunk 4]
```diff
@@ -1,14 +1,30 @@
+import { allArtifactSets, allWeaponBowKeys, allWeaponCatalystKeys, allWeaponClaymoreKeys, allWeaponPolearmKeys, allWeaponSwordKeys, nonTravelerCharacterKeys, WeaponBowKey, WeaponCatalystKey, WeaponClaymoreKey, WeaponPoleArmKey, WeaponSwordKey } from "@genshin-optimizer/consts"
+
 export const allHitModes = ["hit", "avgHit", "critHit"] as const
-export const allRegions = ["mondstadt", "liyue", "inazuma", "sumeru", "fontaine", "natlan", "snezhnaya", "khaenriah"] as const
 export const allAmpReactions = ["vaporize", "melt",] as const
 export const allAdditiveReactions = ["spread", "aggravate"] as const
 export const allArtifactSetCount = [1, 2, 3, 4, 5] as const
+/**
```
Comment (nguyentvan7) [inline apps/frontend/src/app/Types/consts.ts:7]: Do we need to have a separate PR that migrates usages of this to the other version, I assume?

Comment (frzyc) [reply apps/frontend/src/app/Types/consts.ts:7]: eventually, the intention here is to only have a single list for `all...` artifact/weapon/character

**Review by nguyentvan7 — APPROVED** [review]
